tests/events/server/actions/test_cancelling_flow_run.py

```diff
@@ -0,0 +1,237 @@
+
+    flow_run = await flow_runs.read_flow_run(
+        session,
+        super_long_exposure.id,
+    )
+    assert flow_run.state_type == StateType.CANCELLING
+
+
+@pytest.fixture
+def cancel_exposures_that_go_into_a_weirdo_state(
+    take_a_picture: Deployment,
+) -> Automation:
+    return Automation(
+        name="If the exposure is longer than 1 minute, cancel it",
+        trigger=EventTrigger(
+            match_related={
+                "prefect.resource.role": "deployment",
+                "prefect.resource.id": f"prefect.deployment.{take_a_picture.id}",
+            },
+            after={"prefect.flow-run.Running"},
+            expect={"prefect.flow-run.Weirdo"},
+            posture=Posture.Reactive,
+            threshold=0,
+            within=timedelta(minutes=1),
+        ),
+        actions=[actions.CancelFlowRun()],
+    )
+
+
+@pytest.fixture
+def cancel_that_weird_exposure(
+    cancel_exposures_that_go_into_a_weirdo_state: Automation,
+    super_long_exposure: FlowRun,
+) -> TriggeredAction:
+    firing = Firing(
+        trigger=cancel_exposures_that_go_into_a_weirdo_state.trigger,
+        trigger_states={TriggerState.Triggered},
+        triggered=pendulum.now("UTC"),
+        triggering_labels={},
+        triggering_event=ReceivedEvent(
+            occurred=pendulum.now("UTC"),
+            event="prefect.flow-run.Weirdo",
+            resource={
+                "prefect.resource.id": f"prefect.flow-run.{super_long_exposure.id}"
+            },
+            id=uuid4(),
+        ),
+    )
+    return TriggeredAction(
+        automation=cancel_exposures_that_go_into_a_weirdo_state,
+        firing=firing,
+        triggered=firing.triggered,
+        triggering_labels=firing.triggering_labels,
+        triggering_event=firing.triggering_event,
+        action=cancel_exposures_that_go_into_a_weirdo_state.actions[0],
+    )
+
+
+async def test_cancelling_weird_exposure(
+    cancel_that_weird_exposure: TriggeredAction,
+    super_long_exposure: FlowRun,
+    session: AsyncSession,
+):
+    flow_run = await flow_runs.read_flow_run(
+        session,
+        super_long_exposure.id,
+    )
+    assert flow_run.state_type == StateType.RUNNING
+
+    action = cancel_that_weird_exposure.action
+    assert isinstance(action, actions.CancelFlowRun)
+
+    await action.act(cancel_that_weird_exposure)
+
+    session.expunge_all()
+
+    flow_run = await flow_runs.read_flow_run(
+        session,
+        super_long_exposure.id,
+    )
+    assert flow_run.state_type == StateType.CANCELLING
+
+
+async def test_success_event(
+    cancel_that_weird_exposure: TriggeredAction,
+    super_long_exposure: FlowRun,
+):
+    action = cancel_that_weird_exposure.action
+
+    await action.act(cancel_that_weird_exposure)
+    await action.succeed(cancel_that_weird_exposure)
+
+    assert AssertingEventsClient.last
+    (event,) = AssertingEventsClient.last.events
+
+    assert event.event == "prefect-cloud.automation.action.executed"
```

Comment:
I know this is a test, but events like these should be `prefect.automation.action.executed` in OSS, right?

Reply:
Ah good eye.  I have not been changing these yet, and I need to talk with some folks about whether we break compatibility with these event names or not.  I was going to make a separate pass through to change them all at once.